### ISSUE TITLE
Issues after upgrading to version 3

### ISSUE THREAD

**Issue by azgul** [tongueroo/ufo#20]
Hey,

Thanks for the tool :-)

```
$ ufo version
3.0.1
```

I attempted upgrading my project through `$ ufo upgrade3`:
```
$ ufo upgrade3
Upgrading structure of your current project to the new ufo version 3 project structure
Upgraded settings: ufo/settings.yml
mv ufo .ufo
Upgrade complete.
```

Running `$ ufo ship <service name>` results in a seemingly required gem not being installed:
```
$ ufo ship commentr
/Users/martin/.rbenv/versions/2.4.2/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- tilt/erb (LoadError)
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/vendor/render_me_pretty/lib/render_me_pretty/erb.rb:42:in `<top (required)>'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/vendor/render_me_pretty/lib/render_me_pretty.rb:9:in `result'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/setting.rb:34:in `load_file'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/setting.rb:21:in `data'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/docker/builder.rb:54:in `image_name'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/docker/builder.rb:81:in `generate_name'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/docker/builder.rb:76:in `store_full_image_name'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/docker/builder.rb:24:in `build'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/cli.rb:127:in `build_docker'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/cli.rb:50:in `ship'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/command.rb:43:in `dispatch'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/exe/ufo:14:in `<top (required)>'
	from /Users/martin/.rbenv/versions/2.4.2/bin/ufo:23:in `load'
	from /Users/martin/.rbenv/versions/2.4.2/bin/ufo:23:in `<main>'
```

Running `$ gem install tilt` we get a bit onwards, but still no avail:
```
$ ufo ship commentr
/Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/docker/builder.rb:54:in `image_name': undefined method `[]' for nil:NilClass (NoMethodError)
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/docker/builder.rb:81:in `generate_name'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/docker/builder.rb:76:in `store_full_image_name'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/docker/builder.rb:24:in `build'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/cli.rb:127:in `build_docker'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/cli.rb:50:in `ship'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/command.rb:43:in `dispatch'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/exe/ufo:14:in `<top (required)>'
	from /Users/martin/.rbenv/versions/2.4.2/bin/ufo:23:in `load'
	from /Users/martin/.rbenv/versions/2.4.2/bin/ufo:23:in `<main>'
```

Not sure why `setting.data` is nil.

`$ cat ufo/settings.yml`:
```
# More info: http://ufoships.com/docs/ufo-settings/
image: <redacted>.dkr.ecr.eu-west-1.amazonaws.com/commentr
# clean_keep: 30
# ecr_keep: 30

aws_profile_ufo_env_map:
  default: stag

ufo_env_cluster_map:
  default: stag-web
  stag: stag-web
  prod: prod-web
```

Was converted into:

`$ cat .ufo/settings.yml`:
```
---
stag:
  aws_profiles:
  - default
  cluster: stag-web
default:
  cluster: stag-web
prod:
  cluster: prod-web
base:
  image: <redacted>.dkr.ecr.eu-west-1.amazonaws.com/commentr
```

**Comment by azgul** [comment]
Also noticed that `.gitignore` is not updated for the new `.ufo/` path.

`ufo init` still appends
```
.ufo/output
.ufo/data
```
to `.gitignore` though.

**Comment by tongueroo** [comment]
@azgul Thanks so much for the bug report!

Wow, lots of dependency issues with this case.  

RE: dependencies issues
So from your debugging output, it looks like you have already solved the initial dependency issue by `gem install tilt` 👍

Just an explanation of the issue: Ufo is currently submoduling the render_me_pretty gem at vendor/render_me_pretty  (within ufo) and that gem depends on tilt. So that's why tilt is not properly being installed :(  Thought it was fixed in #18 but it's not. Thanks for the bug report. Will fix this.

RE: setttings.data is nil
Think this is because the default env is not `UFO_ENV=development`. Before it was `UFO_ENV=prod`. One of the reasons I decided to bump the tool to version 3. https://github.com/tongueroo/ufo/blob/master/CHANGELOG.md  But the upgrade3 command doesn't change the env. Wasnt sure how to handle this at the time. Thinking, maybe print out a warning that default environment has been changed as part of the `lono upgrade3` command also.

I recommend changing your `.ufo/settings.yml` to the longer environment names manually at ths point.  So maybe:

```yaml
---
staging:
  aws_profiles:
  - default
  cluster: stag-web
default:
  cluster: stag-web
production:
  cluster: prod-web
base:
  image: <redacted>.dkr.ecr.eu-west-1.amazonaws.com/commentr
``` 

The clusters still are in the short name form, which is nice. Then you can set your default UFO_ENV by adding it to your `~/.profile`. Example:

```bash
export UFO_ENV=production # for production, default is now development
```

Hope that helps!  Thanks for the bug reports, they help make the tool better.

**Comment by azgul** [comment]
Thanks for getting back to me so quickly. Ah, right. I've changed `staging` to `development` in your proposal and I wont have to explicitly set `UFO_ENV=staging` when invoking `ufo ship`, thanks!

```
This service commentr does not yet exist in the development cluster.  This deploy will create it.
/Users/martin/.rbenv/versions/2.4.2/lib/ruby/gems/2.4.0/gems/ufo-3.0.1/lib/ufo/ship.rb:317:in `read': No such file or directory @ rb_sysopen - ufo/output/commentr.json (Errno::ENOENT)
```

Seems a path wasn't changed to `.ufo`.

**Comment by tongueroo** [comment]
Great catch. Looking now.

**Comment by tongueroo** [comment]
@azgul released 3.1.0 with the fixes:

* fix container_info to refer to .ufo instead of old ufo folder #20
* fix ufo upgrade3 #20
* add render_me_pretty as gemspec dependency, remove vendor/render_me_pretty submodule #20 #18

Thanks for catching "container_info to refer to .ufo instead of old ufo folder".  Missed it because been testing with existing ECS services.

**Comment by azgul** [comment]
I think there's another issue. It seems to try to deploy to the `development` cluster instead of the defined one.

At least if 
> This service commentr does not yet exist in the development cluster.  This deploy will create it.

is to be trusted :) Because the service is already deployed in the `stag-web` clustser.

**Comment by tongueroo** [comment]
@azgul Interesting, can you post your updated `settings.yml`.  Thanks

**Comment by azgul** [comment]
```
development:
  aws_profiles:
  - default
  cluster: stag-web
default:
  cluster: stag-web
production:
  cluster: prod-web
base:
  image: <redacted>.dkr.ecr.eu-west-1.amazonaws.com/commentr
```

**Comment by tongueroo** [comment]
Release version 3.1.1 which fixes the "It seems to try to deploy to the development cluster instead of the defined one." issue. That was just silly 😁  Been testing with users that only have a development cluster setup. Feel free to re-open if there are still issues.

**Comment by tongueroo** [comment]
@azgul RE: Also noticed that .gitignore is not updated for the new .ufo/ path.
Went ahead and fixed that too 😁 version 3.1.2 has the fix.

**Comment by azgul** [comment]
🎉 works this morning, thanks :)
